diff --git a/node_modules/metro-file-map/src/watchers/FallbackWatcher.js b/node_modules/metro-file-map/src/watchers/FallbackWatcher.js
index ddc052f..edec266 100644
--- a/node_modules/metro-file-map/src/watchers/FallbackWatcher.js
+++ b/node_modules/metro-file-map/src/watchers/FallbackWatcher.js
@@ -49,11 +49,15 @@ class FallbackWatcher extends _AbstractWatcher.AbstractWatcher {
   #changeTimers = new Map();
   #dirRegistry = Object.create(null);
   #watched = Object.create(null);
-  async startWatching() {
+  startWatching() {
     this.#watchdir(this.root);
-    await new Promise((resolve) => {
+    ["src", "assets"].forEach((sourceDir) => {
+      const absoluteSourceDir = _path.default.join(this.root, sourceDir);
+      if (!_fs.default.existsSync(absoluteSourceDir)) {
+        return;
+      }
       recReaddir(
-        this.root,
+        absoluteSourceDir,
         (dir) => {
           this.#watchdir(dir);
         },
@@ -63,13 +67,12 @@ class FallbackWatcher extends _AbstractWatcher.AbstractWatcher {
         (symlink) => {
           this.#register(symlink, "l");
         },
-        () => {
-          resolve();
-        },
+        () => {},
         this.#checkedEmitError,
         this.ignored,
       );
     });
+    return Promise.resolve();
   }
   #register(filepath, type) {
     const dir = _path.default.dirname(filepath);
